Makefile

```diff
@@ -148,25 +144,72 @@ build-test-unit-js:
 		--file ./test/docker-compose.unit-tests-js.yml
 
 .PHONY: build-test-integration
-build-test-integration: build
+build-test-integration:
+	@echo "--- Building integration test images"
 	$(DOCKER_BUILDX_BAKE) \
 		--file ./test/docker-compose.integration-tests.build.yml \
-		python-integration-tests rust-integration-tests
+		python-integration-tests \
+		rust-integration-tests
 
 .PHONY: build-test-e2e
-build-test-e2e: build
-	./pants package ./src/python/e2e-test-runner/e2e_test_runner:pex
+build-test-e2e:
+	@echo "--- Building e2e testing image"
+# Any PEX tagged with `e2e-test-pex` is required for our image. This
+# seems like the most straightforward way of capturing these
+# dependencies at the moment.
+	./pants --tag="e2e-test-pex" package ::
 	$(DOCKER_BUILDX_BAKE) \
 		--file ./test/docker-compose.integration-tests.build.yml \
 		e2e-tests
 
-.PHONY: build-docker-images
-build-docker-images: graplctl build-engagement-view
-	echo "--- Building Docker images"
-	$(DOCKER_BUILDX_BAKE) --file docker-compose.build.yml
+.PHONY: build-engagement-view
+build-engagement-view: ## Build website assets
+	@echo "--- Building the engagement view"
+	$(MAKE) -C src/js/engagement_view build
+	cp -r \
+		"${PWD}/src/js/engagement_view/build/." \
+		"${PWD}/src/rust/grapl-web-ui/frontend/"
 
-.PHONY: build
-build: build-service-pexs build-docker-images ## Build Grapl services
+.PHONY: build-ui
+build-ui: ## Build the web UI
+build-ui: build-engagement-view
+	@echo "--- Building the Web UI"
+	$(DOCKER_BUILDX_BAKE) --file docker-compose.build.yml \
+		grapl-web-ui
+
+.PHONY: build-grapl-services
+build-grapl-services: ## Build all the services for the Grapl SaaS
+build-grapl-services: build-ui
```

Comment:
I guess I'm not fully sure what we gain from indicating all of these here, versus splitting up `docker-compose.build.yml` into local and nonlocal.

We have enough issues remembering to add new services to build_and_upload_containers.sh !

Reply:
That's all going to be addressed in the PR I'm working on now, using the HCL file for `buildx bake`. This was a precursor to that, when I was trying to sort out the best way to build things in `build_and_upload_containers.sh`.